server/http/helloapp/helloapp.go

```diff
@@ -10,94 +10,83 @@ import (
 	"github.com/gorilla/mux"
 	"github.com/pkg/errors"
 
-	pluginapi "github.com/mattermost/mattermost-plugin-api"
+	"github.com/mattermost/mattermost-plugin-api/experimental/oauther"
 
 	"github.com/mattermost/mattermost-plugin-apps/server/apps"
-	"github.com/mattermost/mattermost-plugin-apps/server/configurator"
 	"github.com/mattermost/mattermost-plugin-apps/server/constants"
 	"github.com/mattermost/mattermost-plugin-apps/server/utils/httputils"
 )
 
 const AppSecret = "1234"
 
+const (
+	PathManifest             = "/mattermost-app.json"
+	PathWishInstall          = "/wish/install"
+	PathWishConnectedInstall = "/wish/connected_install"
+	PathOAuth2               = "/oauth2"
+	PathOAuth2Complete       = "/oauth2/complete" // /complete comes from OAuther
+)
+
 type helloapp struct {
-	mm           *pluginapi.Client
-	configurator configurator.Service
+	apps    *apps.Service
+	OAuther oauther.OAuther
 }
 
 func Init(router *mux.Router, apps *apps.Service) {
-	a := helloapp{
-		mm:           apps.Mattermost,
-		configurator: apps.Configurator,
+	h := helloapp{
+		apps: apps,
 	}
 
 	subrouter := router.PathPrefix(constants.HelloAppPath).Subrouter()
-	subrouter.HandleFunc("/mattermost-app.json", a.handleManifest).Methods("GET")
+	subrouter.HandleFunc(PathManifest, h.handleManifest).Methods("GET")
+
+	subrouter.HandleFunc(PathWishInstall, wish(h.handleInstall)).Methods("POST")
+	subrouter.HandleFunc(PathWishConnectedInstall, wish(h.handleConnectedInstall)).Methods("POST")
+
+	subrouter.PathPrefix(PathOAuth2).HandlerFunc(h.handleOAuth).Methods(http.MethodGet, http.MethodPost)
```

Comment:
This is the part I like the less about how to handle URLs within the package. The idea to do it like this was to remove dependencies to gorilla mux. But after trying to use it here, I personally didn't like it much. Any idea to make this work better is welcomed.